src/containers/NewProducts/NewProducts.jsx

```diff
@@ -1,43 +1,57 @@
-import { useState, useEffect } from "react";
-import ProductCard from "../../components/ProductCard/ProductCard";
-import fetchProducts from "../../services/fetchProducts";
-import styles from './NewProducts.module.scss';
+import { useState, useEffect } from 'react'
+import ApiClient from '../../services/apiClient'
+import ProductCard from '../../components/ProductCard/ProductCard'
+import styles from './NewProducts.module.scss'
 
 export default function NewProducts() {
-	const [newProducts, setNewProducts] = useState(null);
-	const [offset, setOffset] = useState(0);
-	const filterString = "is_new";
+  const [newProducts, setNewProducts] = useState([])
+  const [offset, setOffset] = useState(0)
+  const filter = 'is_new'
+  const limit = 4
 
-	const showMoreProducts = () => {
-		fetchProducts(offset, setOffset, newProducts, setNewProducts, filterString);
-	};
+  const showMoreProducts = async () => {
+    const params = { filter, offset, limit }
+    const uri = '/v1/products'
+    try {
+      const { data } = await ApiClient.get(uri, { params })
+      setNewProducts((previousProducts) => [
+        ...previousProducts,
+        ...data.products,
+      ])
+      setOffset((previousOffset) => previousOffset + 4)
+    } catch (error) {
+      console.error('Error: ', error.message)
+    }
+  }
 
-	useEffect(() => {
-		fetchProducts(offset, setOffset, newProducts, setNewProducts, filterString);
-	}, []);
+  useEffect(() => {
+    showMoreProducts()
+  }, [])
 
-
-	return (
-		<div className="container">
-			<h2>Новинки </h2>
-			<div className="row">
-				{newProducts && newProducts.length > 0 ? newProducts.map(function (newProduct) {
-					return (
-						<div className="col-lg-3 col-md-4 col-sm-6 col-12 mb-4" key={newProduct.id}>
-							<div className="rounded flex-column h-100">
-								<ProductCard itemCard= {newProduct}/>
-							</div>
-						</div>
-					);
-				}) : "Завантаження..."
-				}
-			</div>
-			<div className="d-flex justify-content-end">
-				<button onClick={showMoreProducts} className={styles.buttonShowMore}>
-					Показати ще
-				</button>
-			</div>
-		</div>
-	)
+  return (
+    <div className="container">
+      <h2>Новинки </h2>
+      <div className="row">
+        {newProducts && newProducts.length > 0
+          ? newProducts.map((newProduct) => {
+              return (
+                <div
+                  className="col-lg-3 col-md-4 col-sm-6 col-12 mb-4"
+                  key={newProduct.id}
+                >
+                  <div className="rounded flex-column h-100">
+                    <ProductCard itemCard={newProduct} />
+                  </div>
+                </div>
+              )
+            })
+          : 'Завантаження...'}
+      </div>
+      <div className="d-flex justify-content-end">
+        <button onClick={showMoreProducts} className={styles.buttonShowMore}>
+          Показати ще
+        </button>
+      </div>
+    </div>
+  )
 }
```

Comment:
try this instead:
```jsx
import { useState, useEffect } from 'react'
import ApiClient from '../../services/apiClient'
import ProductCard from '../../components/ProductCard/ProductCard'
import styles from './NewProducts.module.scss'

export default function NewProducts() {
  const [products, setProducts] = useState([])
  const [offset, setOffset] = useState(0)
  const [loading, setLoading] = useState(false)
  const filter = 'is_new'
  const limit = 4

  const fetchMoreProducts = async () => {
    const params = { filter, offset, limit }
    const uri = '/v1/products'
    try {
      setLoading(true)
      const { data } = await ApiClient.get(uri, { params })
      setProducts(previousProducts => [...previousProducts, ...data.products])
    } catch (error) {
      console.error('Error: ', error.message)
    } finally {
      setLoading(false)
    }
  }

  const increaseOffset = () => {
    setOffset(previousOffset => previousOffset + limit)
  }

  useEffect(() => {
    if (!loading) {
      fetchMoreProducts()
    }
  }, [offset])

  return (
    <div className='container'>
      <h2>Новинки </h2>
      <div className='row'>
        {products?.length > 0 &&
          products.map(newProduct => {
            return (
              <div className='col-lg-3 col-md-4 col-sm-6 col-12 mb-4' key={newProduct.id}>
                <div className='rounded flex-column h-100'>
                  <ProductCard itemCard={newProduct} />
                </div>
              </div>
            )
          })}
        {loading && 'Завантаження...'}
      </div>
      <div className='d-flex justify-content-end'>
        <button onClick={increaseOffset} className={styles.buttonShowMore}>
          Показати ще
        </button>
      </div>
    </div>
  )
}
```
and plus, return `StrictMode` back in `main.jsx`

Reply:
new proposal- to use: useEffect(() => {
       if (!loading & (productsRef.current === false)) {
      fetchProducts()
        return () => {
        productsRef.current = true
      }
    }
  }, [])